migrator/runner/version_amd64.go

```diff
@@ -0,0 +1,146 @@
+//go:build amd64
+
+package runner
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/migrator/bolthelpers"
+	"github.com/stackrox/rox/migrator/types"
+	"github.com/stackrox/rox/migrator/version"
+	"github.com/stackrox/rox/pkg/migrations"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/tecbot/gorocksdb"
+	bolt "go.etcd.io/bbolt"
+)
+
+var (
+	versionBucketName = []byte("version")
+	versionKey        = []byte("\x00")
+	ctx               = sac.WithAllAccess(context.Background())
+)
+
+// getCurrentSeqNumBolt returns the current seq-num found in the bolt DB.
+// A returned value of 0 means that the version bucket was not found in the DB;
+// this special value is only returned when we're upgrading from a version pre-2.4.
+func getCurrentSeqNumBolt(db *bolt.DB) (int, error) {
+	bucketExists, err := bolthelpers.BucketExists(db, versionBucketName)
+	if err != nil {
+		return 0, errors.Wrap(err, "checking for version bucket existence")
+	}
+	if !bucketExists {
+		return 0, nil
+	}
+	versionBucket := bolthelpers.TopLevelRef(db, versionBucketName)
+	versionBytes, err := bolthelpers.RetrieveElementAtKey(versionBucket, versionKey)
+	if err != nil {
+		return 0, errors.Wrap(err, "failed to retrieve version")
+	}
+	if versionBytes == nil {
+		return 0, errors.New("INVALID STATE: a version bucket existed, but no version was found")
+	}
+	version := new(storage.Version)
+	err = proto.Unmarshal(versionBytes, version)
+	if err != nil {
+		return 0, errors.Wrap(err, "unmarshaling version proto")
+	}
+	return int(version.GetSeqNum()), nil
+}
+
+// getCurrentSeqNumRocksDB returns the current seq-num found in the rocks DB.
+func getCurrentSeqNumRocksDB(db *gorocksdb.DB) (int, error) {
+	var version storage.Version
+
+	opts := gorocksdb.NewDefaultReadOptions()
+	defer opts.Destroy()
+	slice, err := db.Get(opts, versionBucketName)
+	if err != nil || !slice.Exists() {
+		return 0, err
+	}
+	defer slice.Free()
+	if err := proto.Unmarshal(slice.Data(), &version); err != nil {
+		return 0, err
+	}
+	return int(version.GetSeqNum()), nil
+}
+
+func getCurrentSeqNumPostgres(databases *types.Databases) (int, error) {
+	ver, err := version.ReadVersionGormDB(ctx, databases.GormDB)
+	if err != nil {
+		return 0, errors.Wrap(err, "getting current postgres sequence number")
+	}
+
+	return ver.SeqNum, nil
+}
+func getCurrentSeqNum(databases *types.Databases) (int, error) {
+	// If Rocks and Bolt are passed into this function when Postgres is enabled, that means
+	// we are in a state where we need to migrate Rocks to Postgres.  In this case the Rocks
+	// sequence number will take precedence and drive the migrations
```

Comment:
Should the comment be on the function instead of if block?

Also, I am not entirely following the below portion because an error is return in case of mismatched version number.

```
n this case the Rocks
	// sequence number will take precedence and drive the migrations
```